src/Infrastructure.EntityFramework/Auth/Models/AuthRequest.cs

```diff
@@ -7,12 +8,14 @@ public class AuthRequest : Core.Auth.Entities.AuthRequest
 {
     public virtual User User { get; set; }
     public virtual Device ResponseDevice { get; set; }
+    public virtual Organization Organization { get; set; }
 }
 
 public class AuthRequestMapperProfile : Profile
 {
     public AuthRequestMapperProfile()
     {
         CreateMap<Core.Auth.Entities.AuthRequest, AuthRequest>().ReverseMap();
+        CreateMap<Core.Auth.Entities.AuthRequest, OrganizationAdminAuthRequest>();
```

Comment:
I think this would be a better fit as a projection like we did here: 
https://github.com/bitwarden/server/blob/fed0828677428f91c0fc856de1d94c4470ea10ab/src/Infrastructure.EntityFramework/Models/Organization.cs#L29-L35 and consume it like this https://github.com/bitwarden/server/blob/fed0828677428f91c0fc856de1d94c4470ea10ab/src/Infrastructure.EntityFramework/Repositories/OrganizationRepository.cs#L205.

It will hopefully make your query logic simpler too because it should be able to do the join for you.

Reply:
That works out a lot nicer, thanks for pointing me in that direction! I switched it to a projection as you suggested and it simplified the query logic.